srcTest/CompileCheckerTest.java

```diff
@@ -0,0 +1,94 @@
+import java.io.File;
+import static org.junit.Assert.assertTrue;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+/**
+ * Created by stevenroderick on 3/16/16.
+ */
+public class CompileCheckerTest {
+
```

Comment:
For some of these tests, it may help to write a couple helper methods that create and destroy files.  That way, we have control over the data for the tests.